src/conventions/h-group/clue-interpretation/focus-possible.js

```diff
@@ -237,17 +241,15 @@ export function find_focus_possible(state, action) {
 	let focus_possible = [];
 
 	if (clue.type === CLUE.COLOUR) {
-		const colour = state.variant.suits.includes('Rainbow') ? state.variant.suits.indexOf('Rainbow') : clue.value;
-		focus_possible = focus_possible.concat(find_colour_focus(state, colour, action));
+		const colours = state.variant.suits.filter(s => s.match(variantRegexes.rainbowish)).map(s => state.variant.suits.indexOf(s));
+		colours.push(clue.value);
+		colours.forEach(s => focus_possible = focus_possible.concat(find_colour_focus(state, s, action)));
```

Comment:
This can be refactored using Array.flatMap() (combines map() and flat()):
```js
focus_possible = focus_possible.concat(colours.flatMap(colour => find_colour_focus(state, colour, action));
```

I don't remember why I'm starting with an empty array and calling concat(), instead of just assigning. Maybe it's not needed?